TileDB.CSharp/src/TileDBBuffer.cs

```diff
@@ -123,11 +151,98 @@ protected void Release()
             {
                 m_validities.Clear();
             }
+
+    
            
             m_intptr = System.IntPtr.Zero;
         }
         #endregion
 
+        #region string
+        public void PackStringArray(string[] strarray)
+        {
+            Release();
+            if(strarray.Length==0)
+            {
+                return;
+            }
+            m_data = new T[strarray.Length];
+            m_offsets = TileDB.VectorUInt64.Repeat(0, strarray.Length);
+
+            List<byte[]> bytesList = new List<byte[]>();
+
+            int totsize = 0;
+            for(int i=0; i<strarray.Length; ++i)
+            {
+                byte[] bytes = Encoding.ASCII.GetBytes(strarray[i]);
```

Comment:
It may be worth measuring, but I suspect we will be better off doing two passes here:
1. one pass to get the lengths using `Encoding.*.GetByteCount`
2. a single allocation for the output buffer
3. another pass to copy into the output buffer

(1 & 3 can be parallelized -- only if needed in the future for very large data)

Reply:
Thanks for the suggestion. Added one parameter for encoding method, The default method is UTF8. We can also choose other method, such as ASCII, UTF16, and UTF32. As suggested, one of the two copies of the data is removed. Encoding.*.GetByteCount is used to get the total size of the buffer, then the encoded bytes are copied to the output buffer.